ImportError: pikepdf's extension library failed to import

[linuxserverurl]: https://linuxserver.io
[![linuxserver.io](https://raw.githubusercontent.com/linuxserver/docker-templates/master/linuxserver.io/img/linuxserver_medium.png)][linuxserverurl]

<!--- If you are new to Docker or this application our issue tracker is **ONLY** used for reporting bugs or requesting features. Please use [our discord server](https://discord.gg/YWrKVTn) for general support. --->

<!--- Provide a general summary of the bug in the Title above -->

------------------------------

## Expected Behavior
<!--- Tell us what should happen -->
Uploading a document into the paperless-ng web application should trigger an OCR process and the file should be available afterwards.

## Current Behavior
<!--- Tell us what happens instead of the expected behavior -->
The upload process stops at 'Upload complete, waiting...' and nothing happens. The web application itself is fully functional, also the Django admin backend. Just the upload process seems not to work at all.

## Steps to Reproduce
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1. Log into the paperless-ng web application
2. Upload a sample pdf file using the "drop documents here" or "browse file" button
3. Observe that the file is not successfully uploaded or processed

## Environment
**OS:** 
Debian GNU/Linux 9 (stretch) - Raspberry Pi 4

**CPU architecture:** 
arm32

**How docker service was installed:** 
Via portainer, official repo from DockerHub

## Docker logs
<!--- Provide a full docker log, output of "docker logs paperless-ng" -->
````
19:28:16 [Q] INFO recycled worker Process-1:1
19:28:16 [Q] INFO Process-1:5 ready for work at 381
19:28:19 [Q] INFO Enqueued 1
19:28:19 [Q] INFO Process-1:2 processing [example.pdf]
[pid: 347|app: 0|req: 7/7] 192.168.178.7 () {42 vars in 1669 bytes} [Fri Aug 27 19:28:19 2021] POST /api/documents/post_document/ => generated 4 bytes in 33 msecs (HTTP/1.1 200) 10 headers in 292 bytes (2 switches on core 0)
[2021-08-27 19:28:19,256] [INFO] [paperless.consumer] Consuming example.pdf
19:28:19 [Q] INFO Process-1:2 stopped doing work
19:28:19 [Q] ERROR Failed [example.pdf] - pikepdf's extension library failed to import : Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pikepdf/__init__.py", line 13, in <module>
    from . import _qpdf
ImportError: /usr/local/lib/python3.8/dist-packages/pikepdf/_qpdf.cpython-38-arm-linux-gnueabihf.so: undefined symbol: _ZN20QPDFPageObjectHelper16placeFormXObjectE16QPDFObjectHandleRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEENS0_9RectangleEbbb
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/django_q/cluster.py", line 432, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/app/paperless/src/documents/tasks.py", line 74, in consume_file
    document = Consumer().try_consume_file(
  File "/app/paperless/src/documents/consumer.py", line 248, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/app/paperless/src/paperless_tesseract/parsers.py", line 230, in parse
    import ocrmypdf
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/__init__.py", line 10, in <module>
    from ocrmypdf import helpers, hocrtransform, leptonica, pdfa, pdfinfo
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/helpers.py", line 22, in <module>
    import pikepdf
  File "/usr/local/lib/python3.8/dist-packages/pikepdf/__init__.py", line 16, in <module>
    raise ImportError(_msg) from _e
ImportError: pikepdf's extension library failed to import
````

I just tried to import another pdf scanned from switft scanner iOS app and got an error I cant solve on mu RPI4

[2022-01-21 20:42:26,286] [ERROR] [paperless.consumer] Error while consuming document Consulta otorrino António.pdf: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

Traceback (most recent call last):

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 241, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 340, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 359, in run_pipeline

    pdfinfo = get_pdfinfo(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 157, in get_pdfinfo

    return PdfInfo(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 860, in __init__

    self._pages = _pdf_pageinfo_concurrent(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 644, in _pdf_pageinfo_concurrent

    executor(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__

    self._execute(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 132, in _execute

    for result in results:

  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next

    raise value

  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker

    result = (True, func(*args, **kwds))

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 601, in _pdf_pageinfo_sync

    page = PageInfo(pdf, pageno, infile, check_pages, detailed_analysis)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 675, in __init__

    self._gather_pageinfo(pdf, pageno, infile, check_pages, detailed_analysis)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 721, in _gather_pageinfo

    for ci in _process_content_streams(

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 528, in _process_content_streams

    yield from _find_regular_images(container, contentsinfo)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 446, in _find_regular_images

    yield ImageInfo(name=draw.name, pdfimage=pdfimage, shorthand=draw.shorthand)

  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 319, in __init__

    pim_icc = pim.icc

  File "/usr/local/lib/python3.8/dist-packages/pikepdf/models/image.py", line 480, in icc

    self._icc = ImageCms.ImageCmsProfile(iccbytesio)

  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 172, in __init__

    self._set(core.profile_frombytes(profile.read()))

  File "/usr/local/lib/python3.8/dist-packages/PIL/_util.py", line 19, in __getattr__

    raise self.ex

  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 23, in <module>

    from PIL import _imagingcms

ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/app/paperless/src/documents/consumer.py", line 248, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/app/paperless/src/paperless_tesseract/parsers.py", line 290, in parse

    raise ParseError(f"{e.__class__.__name__}: {str(e)}")

documents.parsers.ParseError: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

So after a few more tests it seems to work only on some files, on others not. I tried to find a common property on the files working and the files not working but I didn't find any correlation.

The error now occurring on some files is:
```16:00:33 [Q] INFO Process-1:28 processing [some.pdf]
[2022-04-03 16:00:34,313] [INFO] [paperless.consumer] Consuming some.pdf
[2022-04-03 16:00:37,179] [ERROR] [paperless.consumer] Error while consuming document some.pdf: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)
Traceback (most recent call last):
  File "/app/paperless/src/paperless_tesseract/parsers.py", line 241, in parse
    ocrmypdf.ocr(**args)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 340, in ocr
    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 359, in run_pipeline
    pdfinfo = get_pdfinfo(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 157, in get_pdfinfo
    return PdfInfo(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 860, in __init__
    self._pages = _pdf_pageinfo_concurrent(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 644, in _pdf_pageinfo_concurrent
    executor(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__
    self._execute(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 132, in _execute
    for result in results:
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 601, in _pdf_pageinfo_sync
    page = PageInfo(pdf, pageno, infile, check_pages, detailed_analysis)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 675, in __init__
    self._gather_pageinfo(pdf, pageno, infile, check_pages, detailed_analysis)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 721, in _gather_pageinfo
    for ci in _process_content_streams(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 528, in _process_content_streams
    yield from _find_regular_images(container, contentsinfo)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 446, in _find_regular_images
    yield ImageInfo(name=draw.name, pdfimage=pdfimage, shorthand=draw.shorthand)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 319, in __init__
    pim_icc = pim.icc
  File "/usr/local/lib/python3.8/dist-packages/pikepdf/models/image.py", line 394, in icc
    self._icc = ImageCms.ImageCmsProfile(iccbytesio)
  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 172, in __init__
    self._set(core.profile_frombytes(profile.read()))
  File "/usr/local/lib/python3.8/dist-packages/PIL/_util.py", line 19, in __getattr__
    raise self.ex
  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 23, in <module>
    from PIL import _imagingcms
ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/app/paperless/src/documents/consumer.py", line 248, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/app/paperless/src/paperless_tesseract/parsers.py", line 290, in parse
    raise ParseError(f"{e.__class__.__name__}: {str(e)}")
documents.parsers.ParseError: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)
16:00:37 [Q] INFO Process-1:28 stopped doing work
16:00:37 [Q] ERROR Failed [some.pdf] - some.pdf: Error while consuming document some.pdf: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py) : Traceback (most recent call last):
  File "/app/paperless/src/paperless_tesseract/parsers.py", line 241, in parse
    ocrmypdf.ocr(**args)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/api.py", line 340, in ocr
    return run_pipeline(options=options, plugin_manager=plugin_manager, api=True)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_sync.py", line 359, in run_pipeline
    pdfinfo = get_pdfinfo(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_pipeline.py", line 157, in get_pdfinfo
    return PdfInfo(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 860, in __init__
    self._pages = _pdf_pageinfo_concurrent(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 644, in _pdf_pageinfo_concurrent
    executor(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/_concurrent.py", line 82, in __call__
    self._execute(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/builtin_plugins/concurrency.py", line 132, in _execute
    for result in results:
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 868, in next
    raise value
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 601, in _pdf_pageinfo_sync
    page = PageInfo(pdf, pageno, infile, check_pages, detailed_analysis)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 675, in __init__
    self._gather_pageinfo(pdf, pageno, infile, check_pages, detailed_analysis)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 721, in _gather_pageinfo
    for ci in _process_content_streams(
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 528, in _process_content_streams
    yield from _find_regular_images(container, contentsinfo)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 446, in _find_regular_images
    yield ImageInfo(name=draw.name, pdfimage=pdfimage, shorthand=draw.shorthand)
  File "/usr/local/lib/python3.8/dist-packages/ocrmypdf/pdfinfo/info.py", line 319, in __init__
    pim_icc = pim.icc
  File "/usr/local/lib/python3.8/dist-packages/pikepdf/models/image.py", line 394, in icc
    self._icc = ImageCms.ImageCmsProfile(iccbytesio)
  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 172, in __init__
    self._set(core.profile_frombytes(profile.read()))
  File "/usr/local/lib/python3.8/dist-packages/PIL/_util.py", line 19, in __getattr__
    raise self.ex
  File "/usr/local/lib/python3.8/dist-packages/PIL/ImageCms.py", line 23, in <module>
    from PIL import _imagingcms
ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/asgiref/sync.py", line 288, in main_wrap
    raise exc_info[1]
  File "/app/paperless/src/documents/consumer.py", line 248, in try_consume_file
    document_parser.parse(self.path, mime_type, self.filename)
  File "/app/paperless/src/paperless_tesseract/parsers.py", line 290, in parse
    raise ParseError(f"{e.__class__.__name__}: {str(e)}")
documents.parsers.ParseError: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/django_q/cluster.py", line 432, in worker
    res = f(*task["args"], **task["kwargs"])
  File "/app/paperless/src/documents/tasks.py", line 74, in consume_file
    document = Consumer().try_consume_file(
  File "/app/paperless/src/documents/consumer.py", line 266, in try_consume_file
    self._fail(
  File "/app/paperless/src/documents/consumer.py", line 70, in _fail
    raise ConsumerError(f"{self.filename}: {log_message or message}")
documents.consumer.ConsumerError: some.pdf: Error while consuming document some.pdf: ImportError: cannot import name '_imagingcms' from 'PIL' (/usr/local/lib/python3.8/dist-packages/PIL/__init__.py)

16:00:37 [Q] INFO recycled worker Process-1:28
16:00:37 [Q] INFO Process-1:30 ready for work at 3007
``` 
Besides this probably being a different issue, i find it somewhat interesting other exceptions ocuring during exceptionhandling. I guess that should not be.

--- a/readme-vars.yml
+++ b/readme-vars.yml
@@ -48,6 +48,7 @@ app_setup_block: |
   For convenience this container provides an alias to perform administration management commands. Available administration commands are documented upstream [here](https://paperless-ng.readthedocs.io/en/latest/administration.html) and can be accessed with this container thus: `docker exec -it <container_name> manage <command>`. For example, `docker exec -it paperless manage  document_retagger -tT`.
 # changelog
 changelogs:
+  - { date: "05.09.22:", desc: "Deprecate." }
   - { date: "30.11.21:", desc: "Added in jbig2enc." }
   - { date: "07.10.21:", desc: "Fixed variable that determins if we should run redis locally." }
   - { date: "01.10.21:", desc: "Replaced uwsgi with gunicorn due to websocket issues." }

--- a/readme-vars.yml
+++ b/readme-vars.yml
@@ -6,8 +6,12 @@ project_url: "https://github.com/jonaswinkler/paperless-ng"
 project_logo: ""
 project_blurb: |
   [{{ project_name|capitalize }}]({{ project_url }}) is an application by Daniel Quinn and contributors that indexes your scanned documents and allows you to easily search for documents and store metadata alongside your documents."
-
 project_lsio_github_repo_url: "https://github.com/linuxserver/docker-{{ project_name }}"
+
+project_deprecation_status: true
+project_deprecation_message: |
+  We recommend switching to the new official container: 
+  https://github.com/paperless-ngx/paperless-ngx
 
 # supported architectures
 available_architectures:

